docs/testing/cypress.mdx

```diff
@@ -0,0 +1,193 @@
+> [!NOTE]
+> Check out the [demo repo](https://github.com/clerk/example-cypress-nextjs) that tests a Clerk-powered application using [Testing Tokens](/docs/testing/overview#testing-tokens).
+
+<Steps>
+### Install `@clerk/testing`
+
+Clerk's testing package provides integration helpers for popular testing frameworks. Install it by running the following command:
+
+<CodeBlockTabs options={['npm', 'yarn', 'pnpm']}>
+
+```sh {{ filename: 'terminal' }}
+npm i @clerk/testing --save-dev
+```
+
+```sh {{ filename: 'terminal' }}
+yarn add -D @clerk/testing
+```
+
+```sh {{ filename: 'terminal' }}
+pnpm add @clerk/testing -D
+```
+
+</CodeBlockTabs>
+
+
+### Set your API keys
+
+In your test runner, set your publishable and secret key as the `CLERK_PUBLISHABLE_KEY` and `CLERK_SECRET_KEY` environment variables, respectively.
+
+To find your keys:
+1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=api-keys).
+2. In the navigation sidebar, select **API Keys**.
+3. In the **Quick Copy** section, copy your Clerk publishable and secret key.
+
+> [!WARNING]
+> Ensure you provide the secret key in a secure manner, to avoid leaking it to third parties. For example, if you are using GitHub Actions, refer to [*Using secrets in GitHub Actions*](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions).
+
+
+### Global setup
+
+To set up Clerk with Cypress, add the following code to your [`cypress.config.ts`](https://docs.cypress.io/guides/references/configuration) file.
+
+```tsx filename="cypress.config.ts"
+import { clerkSetup } from '@clerk/testing/cypress';
+import { defineConfig } from 'cypress';
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
+      return clerkSetup({ config });
+    },
+    baseUrl: 'http://localhost:3000' // your app's URL
+  }
+});
+```
+
+`clerkSetup` will obtain a [Testing Token](/docs/testing/overview#testing-tokens) when your test suite starts, making it available for all subsequent tests to use.
+
+### Use Clerk Testing Tokens
+
+Now that Cypress is configured with Clerk, you can use the `setupClerkTestingToken()` function in your tests to augment them with the Testing Token. See the following example:
+
+```tsx filename="my-test.cy.ts"
+import { setupClerkTestingToken } from "@clerk/testing/cypress";
+
+it("sign up", () => {
+  setupClerkTestingToken();
+
+  cy.visit('/sign-up');
+  // ...
+});
+```
+
+</Steps>
+
+## Custom Commands
+
+The `@clerk/testing` package also provides [Cypress custom commands](https://docs.cypress.io/api/cypress-api/custom-commands) to sign in/sign out with Clerk in your Cypress tests without having to interact with the UI.
+To use these commands, you must import them in your [Cypress E2E Support file](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Support-file), which by default is the `cypress/support/e2e.{js,jsx,ts,tsx}` file.
+
+Example:
+```ts {{ filename: 'cypress/support/commands.ts' }}
```

Comment:
File name should match the default

```suggestion
```ts {{ filename: 'cypress/support/e2e.ts' }}
```